backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java

```diff
@@ -77,8 +88,27 @@ public CompletableFuture<Backup> restore(final BackupIdentifier id) {
   }
 
   @Override
-  public CompletableFuture<Void> markFailed(final BackupIdentifier id) {
-    throw new UnsupportedOperationException();
+  public CompletableFuture<BackupStatusCode> markFailed(final BackupIdentifier id) {
+    return setStatus(id, new Status(BackupStatusCode.FAILED, "Explicitly marked as failed"));
+  }
```

Comment:
:thought_balloon: We could also pass the error message as a parameter. BackupManager can then set the reason why it is explicitly marking it as failure.

Reply:
:question: If for any reason the status is already "completed", should we overwrite and mark it as failed?

Reply:
When would the manager actually call this? Depending on the use case, overwriting a "completed" status makes could make sense or not.